### DISCUSSION TITLE
Auth middleware Error.

### DISCUSSION THREAD

**Discussion by Pratikkshirsagar** [adonisjs/core#1651] [General]
Hi,

I'm using the Adonis 5 auth middleware, I have also register the Middleware  inside the start/kernel.ts file.
The issue is that in routes.ts file while using middleware('auth') it is giving me error 

```ts
TS2559: Type '"auth"' has no properties in common with type '{ create?: string | ((ctx: HttpContextContract, next: () => Promise<void>, args?: string[] | undefined) => Promise<void>) | MiddlewareHandler[] | undefined; ... 5 more ...; destroy?: string | ... 2 more ... | undefined; } & { ...; }'.
```

Do you have an idea of what the problem is ?

**Comment by thetutlage** [comment]
Can you atleast format the error properly or share the screenshot of the error. Please don't consider asking question as dumping anything into the editor. If you don't ask your questions properly, then there is no almost zero chances or getting answers for them as well

**Reply by Pratikkshirsagar** [reply]
Sorry missed on formatting I have formatted the code now

**Reply by thetutlage** [reply]
No worries. The goal is to have quality and focused discussions, so that the person coming after you should be able to read these discussions and solve their issues

**Comment by thetutlage** [comment]
How are you using the auth middleware? Can you share the code for `start/routes.ts` file and `start/kernel.ts`?

**Reply by Pratikkshirsagar** [reply]
`start/routes.ts`
```ts
import Route from '@ioc:Adonis/Core/Route';

Route.resource('profiles', 'ProfilesController').middleware('auth');
```
`start/kernel.ts`
```ts

import Server from '@ioc:Adonis/Core/Server';

Server.middleware.register(['Adonis/Core/BodyParserMiddleware']);

Server.middleware.registerNamed({
  auth: 'App/Middleware/Auth',
});

```
the middleware auth is generated during the setup.

**Reply by mastermunj** [reply]
For the resource route, you should use middleware as:

```ts
Route.resource('profiles', 'ProfilesController').middleware({ '*': 'auth' });
```
